src/lib/Preview.js

```diff
@@ -1845,9 +1851,13 @@ class Preview extends EventEmitter {
      * @return {void}
      */
     updateContentInsightsOptions(options) {
+        this.options.features = { ...this.options.features, advancedContentInsights: options };
         if (this.viewer && this.viewer.pageTracker) {
-            this.previewOptions = { ...this.previewOptions, contentInsights: options };
-            this.viewer.pageTracker.updateOptions(this.previewOptions.contentInsights);
+            this.previewOptions = {
+                ...this.previewOptions,
+                features: { ...this.previewOptions.features, advancedContentInsights: options },
+            };
+            this.viewer.pageTracker.updateOptions(this.previewOptions.features.advancedContentInsights);
```

Comment:
Can't we just use `options` as the argument to the function here?

Reply:
can you explain why we should do so ?

Reply:
nitpick: I also found it a bit difficult to grasp having both `options` and `this.options`

Reply:
I still find this confusing as well. Why not just call `this.viewer.pageTracker.updateOptions(newOptions)` here? It's much easier to read in my option and stays more consistent with the rest of this code block